### DISCUSSION TITLE
How to run command with dynamic sub command

### DISCUSSION THREAD

**Discussion by jokiefer** [tox-dev/tox#3157] [Q&A] [answered]
Hi folks,

i tried to install the specific gdal version, depending on the installed os gdal binaries.

```tox
commands_pre =
    pip install gdal=="`gdal-config --version`.*"
```

But tox will escape the command like:

```bash
py311-django41: commands_pre[0]> pip install 'gdal==`gdal-config --version`.*'
ERROR: Invalid requirement: 'gdal==`gdal-config --version`.*'

```

Why does tox surround the package with ' chars?

**Comment by jokiefer** [comment]
currently i developed a workaround for me:

run 
```bash
echo "gdal==`gdal-config --version`.*" >> requirements-gdal.txt
```

in front of the tox command and add the requirements-gdal.txt to the deps:

```tox
deps=
    -rrequirements-gdal.txt
```

for my ci jobs this worked. But the initial question is not answered.

**Comment by gaborbernat** [comment, marked as answer]
Tox does not support shell command substitution (backticks or `$(...)`) — commands are not run through a shell by default. Tox passes arguments directly to the executable.

Your workaround of writing a requirements file is a valid approach. Alternatively:

1. Use a shell wrapper: `commands_pre = bash -c 'pip install gdal==$(gdal-config --version).*'` (and add `bash` to `allowlist_externals`)
2. Use the [tox-backtocks](https://github.com/Anvil/tox-backtocks) plugin which adds backtick substitution support
3. Use a helper script that resolves the version and installs
